source/agora/node/Config.d

```diff
@@ -482,6 +412,63 @@ public struct DNSConfig
+        ensure(this.address.length > 0, "registry is enabled but no `address` is provided");
+        ensure(this.port > 0, "registry.port: 0 is not a valid value");
+        ensure(this.authoritative.length > 0, "registry: No authoritative zones provided");
+
+        foreach (idx, zone; this.authoritative)
+        {
+            auto rng = zone.name.splitter('.');
+            ensure(!rng.empty, "registry.authoritative: Empty array entry at index {}", idx);
+            ensure(rng.front.length > 0,
+                   "registry.authoritative: Value '{}' at index '{}' starts with a dot ('.')," ~
+                   " which is not allowed. Remove it.", zone.name, idx);
+
+            do {
+                // It might be the empty label, in which case it needs to be last
+                if (rng.front.length == 0)
+                {
+                    rng.popFront();
+                    ensure(rng.empty,
+                           "registry.authoritative: Value '{}' at index '{}' contains" ~
+                           " an empty label, which is not allowed. Remove the double dot.",
+                           zone.name, idx);
+                    break;
+                }
+                ensure(rng.front.length <= 63,
+                       "registry.authoritative: Value '{}' at index '{}' contains a label ('{}') " ~
+                       "which is longer than 63 characters ({} characters), which is not allowed.",
+                       zone.name, idx, rng.front, rng.front.length);
+                rng.popFront();
+            } while (!rng.empty);
+
+            // Now validate the durations are consistent with one another
+            ensure(zone.refresh <= zone.expire,
+                   "registry.authoritative: Zone '{}' field 'refresh' ({}) should be lower than field 'expire' ({})",
+                   zone.name, zone.refresh, zone.expire);
+            // The other ones could actually be set to very low values to
+            // avoid clients caching data, so don't validate them besides
+            // checking they fit in an `int`.
+            const intMaxSecs = int.max.seconds;
+            const uintMaxSecs = uint.max.seconds;
+            ensure(zone.refresh <= intMaxSecs,
+                   "registry.authoritative: Zone '{}' field 'refresh' ({}) should be at most {}",
+                   zone.name, zone.refresh, intMaxSecs);
+            ensure(zone.retry <= intMaxSecs,
+                   "registry.authoritative: Zone '%s' field 'retry' ({}) should be at most {}",
+                   zone.name, zone.retry, intMaxSecs);
+            ensure(zone.expire <= intMaxSecs,
+                   "registry.authoritative: Zone '%s' field 'expire' ({}) should be at most {}",
+                   zone.name, zone.expire, intMaxSecs);
+            ensure(zone.minimum <= intMaxSecs,
+                   "registry.authoritative: Zone '%s' field 'minimum' ({}) should be at most {}",
+                   zone.name, zone.minimum, uintMaxSecs);
```

Comment:
```suggested
                   zone.name, zone.minimum, intMaxSecs);
```

Reply:
Good eye! Although the error is actually on the comparison, as `minimum` is unsigned:

> MINIMUM         The unsigned 32 bit minimum TTL field that should be
>                           exported with any RR from this zone.

[Source](https://datatracker.ietf.org/doc/html/rfc1035#section-3.3.13).
Added a commit to fix it.